src/main/java/ru/itmo/sd/deadliner2bot/state/AddStartDateState.java

```diff
@@ -0,0 +1,75 @@
+package ru.itmo.sd.deadliner2bot.state;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.meta.api.methods.BotApiMethod;
+import ru.itmo.sd.deadliner2bot.messages.MessageService;
+import ru.itmo.sd.deadliner2bot.model.Chat;
+import ru.itmo.sd.deadliner2bot.model.ChatStateEnum;
+import ru.itmo.sd.deadliner2bot.model.Todo;
+import ru.itmo.sd.deadliner2bot.repository.ChatRepository;
+import ru.itmo.sd.deadliner2bot.service.TodoService;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class AddStartDateState implements ChatState {
+
+    private final ChatRepository chatRepository;
+    private final ChatStateEnum chatStateEnum = ChatStateEnum.ADD_START_DATE;
+    private final MessageService messageService;
+    private final TodoService todoService;
+    //    private final String dateFormat = "yyyy-MM-dd HH:mm:ss z";
+    private final String dateFormat = "dd-MM-yyyy";
+    private final String timeFormat = "HH:mm";
+
+
+    @Override
+    public List<BotApiMethod<?>> process(Chat chat, String message) {
+        Optional<Todo> todo = todoService.findSelectedTodoByChatId(chat.getChatId());
+        if (message.startsWith("/cancel")) {
+            chat.setState(ChatStateEnum.BASE_STATE);
+            chatRepository.save(chat);
+            if (todo.isPresent()) {
+                todo.get().setSelected(false);
+                todoService.save(todo.get());
+            }
+            return List.of(messageService.createMessage(chat, "Operation canceled."));
+        } else {
+            if (todo.isEmpty()) {
+                chat.setState(ChatStateEnum.BASE_STATE);
+                chatRepository.save(chat);
+                return List.of(messageService.createMessage(chat, "No todo selected, cancelled."));
```

Comment:
Actually, this (and similar cases in other states) looks like something extremely odd, since our backend only controls user states. It is a precondition that we have TODO selected.
Could you add logging at the warning level here and other similar places?